service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -231,14 +232,22 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
   @Override
   public ResponseEntity<AbortRunSetResponse> abortRunSet(UUID runSetId) {
     AbortRunSetResponse aborted = new AbortRunSetResponse();
-    List<UUID> dummyRunsList = new ArrayList<>();
+    aborted.runSetId(runSetId);
 
-    dummyRunsList.add(UUID.randomUUID());
-    dummyRunsList.add(UUID.randomUUID());
+    List<Run> runningWorkflows =
+        runDao.getRuns(new RunDao.RunsFilters(runSetId, NON_TERMINAL_STATES));
+    List<String> submittedAbortWorkflows = new ArrayList<>();
 
-    aborted.runSetId(runSetId);
+    for (Run run : runningWorkflows) {
+      try {
+        cromwellService.submitAbortWorkflow(run);
+        submittedAbortWorkflows.add(run.engineId());
+      } catch (cromwell.client.ApiException e) {
+        log.error("Unable to abort workflow %s".formatted(run.engineId()), e);
```

Comment:
To think about, do we want to:
- Return this error (or perhaps a set of errors) to the user?
- Let the user know that not everything went according to plan?
- Mark these errors against the workflows?

Reply:
By the last bullet point, do you mean adding the message to the `errors` columns in the db?